distributed/scheduler.py

```diff
@@ -5543,9 +5858,14 @@ def transition(self, key, finish, *args, **kwargs):
                     "Impossible transition from %r to %r" % (start, finish)
                 )
 
+            for worker, msg in worker_msgs.items():
+                self.worker_send(worker, msg)
+            for client, msg in client_msgs.items():
+                self.client_send(client, msg)
```

Comment:
It might not help any, but any thoughts on pulling these out of the `transition` function into `transitions` ?

Reply:
It looks like they're a decent chunk of the current time spent

![image](https://user-images.githubusercontent.com/306380/105418263-ad705c00-5bf1-11eb-8961-c5ae760fcfb4.png)

Reply:
It's an interesting suggestion and agree it's worth trying. That said, I feel like this PR has gotten quite long and am worried that has slowed down the reviewing of it. Instead would propose we focus on getting this in. That would make it easier to make smaller more focused PRs on moving communication further out and consolidating messages further as well as allowing us to more effectively Cythonize any remaining functions we still see popping up in the profiles. Thoughts? 🙂

(Should add if we want to do this after the patch release tomorrow that also makes sense to me)

Reply:
This was mostly done in PR ( https://github.com/dask/distributed/pull/4451 ). Handling the remaining ones in PR ( https://github.com/dask/distributed/pull/4526 )